src/tools/AssetScanner.js

```diff
@@ -0,0 +1,281 @@
+const path = require('path');
+const fs = require('fs');
+const promisify = require('util').promisify;
+const imageSize = require('image-size');
+const musicMetadata = require('music-metadata');
+
+const { parseSRASConfig } = require('../lib/srasConfig');
+const { formatMsToHRT } = require('../lib/format');
+const { formatHRTFileSize } = require('../lib/format');
+const { isDirectory } = require('../lib/path-validation.js');
+const { isPowerOfTwo } = require('../lib/mathHelpers.js');
+
+const fsReadDir = promisify(fs.readdir);
+const fsStat = promisify(fs.stat);
+const sizeOf = promisify(imageSize);
+
+const resultDefaults = {
+  time: {
+    start: 0,
+    end: 0,
+    get duration() {
+      return this.end - this.start;
+    },
+  },
+  files: {
+    skipped: 0,
+    ignored: 0,
+    scanned: 0,
+  },
+  reports: [],
+};
+
+const scannedTypes = {};
+const unpermittedTypes = {};
+const scanReports = [];
+
+let results;
```

Comment:
I could see an issue where these global variables cause problems: Multiple runs would cause this `results` variable to contain the data from all of the runs. Do we have a process for resetting the variable? Or... Should we extract this into a class?